0006-bingings.md

```diff
@@ -0,0 +1,591 @@
+(that originally initiated the chain of cross-contract calls, the immediate contract that called the current one, the account of the current contract),
+other important information like storage usage.
+
+Many of the below functions are currently implemented through `data_read` which allows to read generic context data.
+However, there is no reason to have `data_read` instead of the specific functions:
+* `data_read` does not solve forward compatibility. If later we want to add another context function, e.g. `executed_operations`
+we can just declare it as a new function, instead of encoding it as `DATA_TYPE_EXECUTED_OPERATIONS = 42` which is passed
+as the first argument to `data_read`;
+* `data_read` does not help with renaming. If later we decide to rename `initiator_id` to `originator_id` then one could
+argue that contracts that rely on `data_read` would not break, while contracts relying on `initiator_id()` would. However
+the name change often means the change of the semantics, which means the contracts using this function are no longer safe to
+execute anyway.
+
+However there is one reason to not have `data_read` -- it makes `API` more human-like which is a general direction Wasm APIs, like WASI are moving towards to.
+
+---
+```rust
+initiator_id(register_id: u64)
+```
+All contract calls are a result of some transaction that was signed by some account using
+some access key and submitted into a memory pool (either through the wallet using RPC or by a node itself). This function returns the id of that account.
+
+###### Normal operation
+* Saves the bytes of the initiator account id into the register.
+
+###### Panics
+* If the scratch buffer size exceeds the limit panics with `MemoryAccessViolation`;
+
+###### Current bugs
+* Currently we conflate `originator_id` and `sender_id` in our code base.
+
+---
+```rust
+initiator_key(register_id: u64)
+```
+Saves JSON-serialized access key that was used by the initiator into the register.
+In rare situations smart contract might want to know the exact access key that was used to send the original transaction,
+e.g. to increase the allowance or manipulate with the public key.
+
+###### Normal operation
+* Saves the JSON-serialized access key into the buffer.
+
+###### Panics
+* If the scratch buffer size exceeds the limit panics with `MemoryAccessViolation`;
+
+
+###### Current bugs
+* Not implemented.
+
+---
+```rust
+caller_id(register_id: u64)
+```
+All contract calls are a result of a receipt, this receipt might be created by a transaction
+that does function invocation on the contract or another contract as a result of cross-contract call.
+
+###### Normal operation
+* Saves the bytes of the caller account id into the register.
+
+###### Panics
+* If the scratch buffer size exceeds the limit panics with `MemoryAccessViolation`;
+
+###### Current bugs
+* Not implemented.
+
+---
+```rust
+refund_id(register_id: u64)
```

Comment:
`refund_account_id` makes more sense

Reply:
Refund account is now irrelevant with new balance model.